lib/serverpipe.js

```diff
@@ -36,6 +36,23 @@ export function fetchWithCredentials(url, options = {}) {
     );
 }
 
+export function getJsonWithCredentials(url, body, options = {}) {
+    const defaultHeaders = {
+        Accept: 'application/json',
+        'Content-Type': 'application/json',
+    };
+    const headers = 'headers' in options ? [...options.headers, defaultHeaders] : defaultHeaders;
+
+    headers[getCsrfHeaderName()] = getCsrfToken();
```

Comment:
For a GET?

Reply:
yes

Reply:
The CSRF filter shouldn't execute for GETs?

Reply:
should we? maybe other server end could be checking csrf header? (this is extracted from other project btw..)

Reply:
A GET endpoint shouldn't ever change state, so there should no risk of a cross-site request causing damage.

Reply:
🌊